File: Java/TirePressureMonitoringSystem/src/main/java/tddmicroexercises/tirepressuremonitoringsystem/Alarm.java
package tddmicroexercises.tirepressuremonitoringsystem;

public class Alarm {

    private Sensor sensor;
    private boolean alarmOn = false;

    public Alarm(Sensor sensor) {
        this.sensor = sensor;
    }

    public void check() {
        alarmOn = !withinBounds(sensor.popNextPressurePsiValue());
    }

    public boolean isAlarmOn() {
        return alarmOn;
    }

    private boolean withinBounds(double psiPressureValue) {
        return (double) 17 <= psiPressureValue && psiPressureValue <= (double) 21;
    }
}

File: kotlin/TirePressureMonitoringSystem/src/main/kotlin/tddmicroexercises/tirepressuremonitoringsystem/Alarm.kt
package tddmicroexercises.tirepressuremonitoringsystem

class Alarm(private val sensor: Sensor) {
    private val LowPressureThreshold = 17.0
    private val HighPressureThreshold = 21.0

    var isAlarmOn = false
        internal set

    fun check() {
        isAlarmOn = sensor.popNextPressurePsiValue() !in LowPressureThreshold..HighPressureThreshold
    }

}
